library/core/src/main/java/com/google/android/exoplayer2/ExoPlayerImpl.java

```diff
@@ -1265,7 +1265,7 @@ private void updatePlaybackInfo(
                 .windowIndex;
         mediaItem = newPlaybackInfo.timeline.getWindow(windowIndex, window).mediaItem;
       }
-      newMediaMetadata = mediaItem != null ? mediaItem.mediaMetadata : MediaMetadata.EMPTY;
+      mediaMetadata = mediaItem != null ? mediaItem.mediaMetadata : MediaMetadata.EMPTY;
```

Comment:
I think you're reintroducing another bug here. It happens to fix the bug you're seeing, but this line really _should_ be assigning to `newMediaMetadata`.

The real bug is that this line correctly updates `newMediaMetadata`, but then L1271-1272 below don't execute to inject in the metadata from the ID3 data.

This can be fixed by by changing the condition on L1270 to be:

```
if (mediaItemTransitioned
    || !previousPlaybackInfo.staticMetadata.equals(newPlaybackInfo.staticMetadata)) {
```

The fix is easy enough that it's probably quicker for us to do it internally, so I'll close this PR. Thanks!